framework/LinePlot/LinePlot.swift

```diff
@@ -62,50 +62,71 @@ public class LineGraph : Plot {
     let s = Series(points: pts, label: l, color: c)
     series.append(s)
   }
+  public func addSeries(function f : (Float)->Float, x1 a : Float, x2 b : Float, numberOfSamples numberOfSamples : Int = 400, label l: String, color c : Color = Color.lightBlue) {
```

Comment:
Generally in Swift, parameter names tend to align with their internal names unless the signature can be defined in a better way by having them be different. Also, I tend to like having descriptive variable names. Single-character variable names are something I tend to avoid.

When it comes to whitespace around colons, the [Google Style Guide](https://google.github.io/swift/) and standard Swift style is to have no whitespace before the colon, and one space afterward for parameters. In my projects, you might notice that I eliminate whitespace before and after colons, but that's my own idiosyncratic style that I'm gradually working myself out of. I lost that argument with others in the Swift community early on.

I might suggest a function definition like the following:
```
public func addSeries(function:(Float)->Float, xMin: Float, xMax: Float, numberOfSamples: Int = 400, label: String, color: Color = Color.lightBlue) {
```

with the appropriate modification in the following function to support those internal parameter names. This is something that could be applied across all functions, and I think it would make them both easier to read and easier to understand the internal logic.

Reply:
Also, would addFunction() be a better name than addSeries(), in order to set this apart? In that case, you maybe could change addSeries to:

```
public func addSeries(_ series: Series){
```

and addFunction to:

```
public func addFunction(_ function: (Float)->Float, xMin: Float, xMax: Float, numberOfSamples: Int = 400, label: String, color: Color = Color.lightBlue) {
```

making the first arguments implicit in the title of the function.

Reply:
Now that I think about it, addFunction does seem better.

Reply:
I'm sorry about this, I know you had mentioned it earlier. I have this bad habit of writing non-descriptive variable names and later get confused when I revisit the code. I'll take extra and deliberate care from now on.